packages/pn-pa-webapp/src/pages/ApiKeys.page.tsx

```diff
@@ -85,14 +95,32 @@ const ApiKeys = () => {
   const { t } = useTranslation(['apikeys']);
 
   const apiKeys = useAppSelector((state: RootState) => state.apiKeysState.apiKeys);
+  const pagination = useAppSelector((state: RootState) => state.apiKeysState.pagination);
+
+  const totalElements =
+    pagination.size *
+    (pagination.moreResult
+      ? pagination.nextPagesKey.length + 5
+      : pagination.nextPagesKey.length + 1);
+  const pagesToShow: Array<number> = calculatePages(
+    pagination.size,
+    totalElements,
+    Math.min(pagination.nextPagesKey.length + 1, 3),
+    pagination.page + 1
+  );
 
   const fetchApiKeys = useCallback(() => {
-    void dispatch(getApiKeys());
-  }, []);
+    const params = {
+      limit: pagination.size,
+      lastKey: pagination.page === 0 ? undefined : pagination.nextPagesKey[pagination.page - 1],
```

Comment:
In the redux slice, store lastKey and lastUpdate. See the comment below, in the reducer file

Reply:
changes made into 5eb41f24bacd7862f6c9ac99ae9f9e9396e71a40